src/test/java/uk/gov/companieshouse/web/pps/util/PenaltyUtilsTest.java

```diff
@@ -26,4 +29,40 @@ void testGetReferenceTitle(){
         assertEquals("Reference Number", result);
     }
 
+    @Test
+    void testGetLoginEmailSuccessful() {
+        String email = "test@gmail.com";
+        Map<String, Object> userProfile = Map.of("email", email);
+        Map<String, Object> signInInfo = Map.of("user_profile", userProfile);
+        SessionService sessionService = new SessionService() {
+            @Override
+            public Map<String, Object> getSessionDataFromContext() {
+                return Map.of("signin_info", signInInfo);
+            }
+        };
+        assertEquals(email, penaltyUtils.getLoginEmail(sessionService));
+    }
+
+    @Test
+    void testGetLoginEmailSuccessful_NullSignInInfo() {
+        SessionService sessionService = new SessionService() {
+            @Override
+            public Map<String, Object> getSessionDataFromContext() {
+                return Map.of("id", "test");
+            }
+        };
+        assertNull(penaltyUtils.getLoginEmail(sessionService));
+    }
+
+    @Test
+    void testGetLoginEmailSuccessful_NullUserProfile() {
+        Map<String, Object> signInInfo = Map.of("id", "test");
+        SessionService sessionService = new SessionService() {
```

Comment:
This can be replaced with a lambda: 
`SessionService sessionService = () -> Map.of("signin_info", signInInfo);`